evaluation/gaia/run_infer.py

```diff
@@ -0,0 +1,336 @@
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setFormatter(
+            logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+        )
+        logger.addHandler(file_handler)
+
+    logger.info(f'Process-specific workspace mounted at {workspace_mount_path}')
+    sandbox = DockerSSHBox()
+    if instance['file_name'] != '':
+        # if this question comes with a file, we need to save it to the workspace
+        src_file = os.path.join(
+            DATASET_CACHE_DIR, '2023', metadata['data_split'], instance['file_name']
+        )
+        extension_name = instance['file_name'].split('.')[-1]
+        dest_file = os.path.join(workspace_mount_path, f'file.{extension_name}')
+        shutil.copyfile(src_file, dest_file)
+        logger.info(f'File copied to {dest_file}')
+    else:
+        dest_file = None
+
+    # Prepare instruction
+    instruction = f"{instance['Question']}"
+    logger.info(f'Instruction: {instruction}')
+    if dest_file:
+        instruction += f"\n\nThe mentioned file is provided in the workspace at: {dest_file.split('/')[-1]}"
+
+    instruction += 'IMPORTANT: You should ONLY interact with the environment provided to you AND NEVER ASK FOR HUMAN HELP.\n'
+    # NOTE: You can actually set slightly different instruction for different agents
+    instruction += AGENT_CLS_TO_INST_SUFFIX.get(agent_class, '')
+
+    # Here's how you can run the agent (similar to the `main` function) and get the final task state
+    state: State = asyncio.run(
+        main(
+            instruction,
+            fake_user_response_fn=AGENT_CLS_TO_FAKE_USER_RESPONSE_FN.get(agent_class),
+            sandbox=sandbox,
+        )
+    )
+    # ======= Attempt to evaluate the agent's edits =======
+    # If you are working on simplier benchmark that only evaluates the final model output (e.g., in a MessageAction)
+    # You can simply get the LAST `MessageAction` from the returned `state.history` and parse it for evaluation.
+
+    if state is None:
+        raise ValueError('State should not be None.')
+
+    model_answer = ''
+    for act, _ in reversed(state.history):
+        if isinstance(act, MessageAction) and act.source == 'model':
+            model_answer = act.content
+            break
+    logger.info(
+        f'Final message: {model_answer} | Ground truth: {instance["Final answer"]}'
+    )
+    test_result = question_scorer(
+        model_answer=model_answer, ground_truth=instance['Final answer']
+    )
+
+    # Save the output
+    output = {
+        'instance_id': instance['task_id'],
+        'instance': instance,
+        'instruction': instance['Question'],
+        'metadata': metadata,
+        'history': [
+            (event_to_dict(action), event_to_dict(obs)) for action, obs in state.history
+        ],
+        'error': state.error if state and state.error else None,
+        'test_result': test_result,
+        'model_answer': model_answer,
```

Comment:
Maybe you can make 'test_result' a dict, and put the `bool` returned from `question_scorer` into it, as well as `model_answer` and `ground_truth` (just for visualization purpose)? So we don't need an `model_answer` on the top-level